oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/util/ThrottlingDocumentStoreWrapper.java

```diff
@@ -0,0 +1,216 @@
+        performThrottling();
+        return store.remove(collection, indexedProperty, startValue, endValue);
+    }
+
+    @Override
+    public <T extends Document> boolean create(final Collection<T> collection, final List<UpdateOp> updateOps) {
+        performThrottling();
+        return store.create(collection, updateOps);
+    }
+
+    @Override
+    public <T extends Document> T createOrUpdate(final Collection<T> collection, final UpdateOp update) {
+        performThrottling();
+        return store.createOrUpdate(collection, update);
+    }
+
+    @Override
+    public <T extends Document> List<T> createOrUpdate(final Collection<T> collection, final List<UpdateOp> updateOps) {
+        performThrottling();
+        return store.createOrUpdate(collection, updateOps);
+    }
+
+    @Override
+    public <T extends Document> T findAndUpdate(final Collection<T> collection, final UpdateOp update) {
+        performThrottling();
+        return store.findAndUpdate(collection, update);
+    }
+
+    @Override
+    public CacheInvalidationStats invalidateCache() {
+        return store.invalidateCache();
+    }
+    
+    @Override
+    public CacheInvalidationStats invalidateCache(Iterable<String> keys) {
+        return store.invalidateCache(keys);
+    }
+
+    @Override
+    public <T extends Document> void invalidateCache(Collection<T> collection, String key) {
+        store.invalidateCache(collection, key);
+    }
+
+    @Override
+    public void dispose() {
+        store.dispose();
+    }
+
+    @Override
+    public <T extends Document> T getIfCached(final Collection<T> collection, final String key) {
+        return store.getIfCached(collection, key);
+    }
+
+    @Override
+    public void setReadWriteMode(String readWriteMode) {
+        store.setReadWriteMode(readWriteMode);
+    }
+
+    @Override
+    public Iterable<CacheStats> getCacheStats() {
+        return store.getCacheStats();
+    }
+
+    @Override
+    public Map<String, String> getMetadata() {
+        return store.getMetadata();
+    }
+
+    @NotNull
+    @Override
+    public Map<String, String> getStats() {
+        return store.getStats();
+    }
+
+    @Override
+    public long determineServerTimeDifferenceMillis() {
+        return store.determineServerTimeDifferenceMillis();
+    }
+
+    /**
+     * Return the size limit for node name based on the document store implementation
+     *
+     * @return node name size limit
+     */
+    @Override
+    public int getNodeNameLimit() {
+        return store.getNodeNameLimit();
+    }
+
+    /**
+     * Return the {@link Throttler} for the underlying store
+     * Default is no throttling
+     *
+     * @return throttler for document store
+     */
+    @Override
+    public Throttler throttler() {
+        return store.throttler();
+    }
+
+    // helper methods
+
+    private void performThrottling() {
+
```

Comment:
I think we should not throttle the lease updates - even though currently the throttling times are lowish and probably don't interfere with the 2+ min lease durations. If the lease cannot be maintained, the throttling is equivalent to a shutdown. And that's probably not the intention. So, even though not throttling lease updates increases the use of oplog window while that is low, I'd argue lease updates are a high priority kind of data and could be excempted.

One way to do that - with lease line changes - could for example be to change `performThrottling` like so:
```
    private void performThrottling(final Collection<T> collection) {
        if (Collection.CLUSTER_NODES == collection) {
            // no throttling of CLUSTER_NODES collection
            return;
        }
   ...
```

Reply:
I think it makes sense to exempt lease updates and other high-priority updates (if any).